src/cobalt_model/frontmatter.rs

```diff
@@ -425,11 +426,19 @@ impl FrontmatterBuilder {
             permalink
         };
 
-        if let Some(ref tags) = tags {
-            if tags.iter().any(|x| x.trim().is_empty()) {
-                return Err("Empty strings are not allowed in tags".into());
+        let tags = if let Some(tags) = tags {
+            if tags.is_empty() {
+                None
+            } else {
+                if tags.iter().any(|x| x.trim().is_empty()) {
+                    return Err("Empty strings are not allowed in tags".into());
+                }
+                Some(tags)
             }
-        }
+        } else {
+            // default value
+            Some(vec![DEFAULT_TAGS.to_owned()])
```

Comment:
Two questions
- If we should have built-in support for a default tag or not.
- Is this the right place for it

So far, cobalt gives you a base set of features and you then opt-in for more.  I somewhat lean towards tags being the same way.  We should discuss this on #549

since this is generic tag behavior and not pagination, shouldn't this be its own PR?

As for if this is the right place or not: this will cause the default to be applied to generic pages and not just posts.   At the moment, that basically is us advertising a feature that does nothing.  I feel like it'd be better if it was on posts only.  Thoughts?

Reply:
> If we should have built-in support for a default tag or not.
https://github.com/cobalt-org/cobalt.rs/issues/550#issuecomment-440687841 ;)

> Is this the right place for it

Do you mean "code wise" or "PR wise"?

> I feel like it'd be better if it was on posts only. 

I didn't realise that here would impact generic pages as well. I agree it should be only for posts, at least for now, I don't see use case to tags pages. But I'm biased by my simple usage of Cobalt.

> since this is generic tag behavior and not pagination, shouldn't this be its own PR? 

I can do another PR for that if you want.

Reply:
> > Is this the right place for it
> Do you mean "code wise" or "PR wise"?

Code wise.  The section I meant as a follow up to this was

> As for if this is the right place or not: this will cause the default to be applied to generic pages and not just posts. At the moment, that basically is us advertising a feature that does nothing. I feel like it'd be better if it was on posts only. Thoughts?

Reply:
Ok, I think I understand better how defaults are managed, basically, it's when we `cobalt new` so if the user wants a default tag, he just needs to put it in the configuration.

My fix for this discussion here is to remove the default alltogether:
```rust
let tags = if let Some(tags) = tags {
            if tags.is_empty() {
                None
            } else {
                if tags.iter().any(|x| x.trim().is_empty()) {
                    return Err("Empty strings are not allowed in tags".into());
                }
                Some(tags)
            }
        } else {
            None // here the change
        };
```

Reply:
Defaults aren't about `cobalt new`.

When we load a page, we load
- the page's frontmatter
- the collection's frontmatter
- the global frontmatter

and merge them.  I'd recommend putting something in the defaults in `_cobalt.yml` and see how it impacts your site.  It is meant to be a simpler approach than jekyll's [default frontmatters](https://jekyllrb.com/docs/configuration/front-matter-defaults/) (no path-based configuration; maybe one day we'll need it).

I'd like to add in a debug flag to showcase what your frontmatter looks like after all of this. I think I want the ECS first since it will make it easier.